File: apps/mobile-app/src/db/models/DetalleDevolucion.ts
import { Model } from '@nozbe/watermelondb';
import { field, date, readonly } from '@nozbe/watermelondb/decorators';

/**
 * Linea individual de una DevolucionPedido. CASCADE on parent.
 * Cantidad y precio congelados al momento de la devolucion (no se recalculan).
 */
export default class DetalleDevolucion extends Model {
  static table = 'detalle_devoluciones';

  @field('server_id') serverId!: number | null;
  @field('devolucion_id') devolucionId!: string;
  @field('devolucion_server_id') devolucionServerId!: number | null;
  @field('detalle_pedido_id') detallePedidoId!: string | null;
  @field('detalle_pedido_server_id') detallePedidoServerId!: number | null;
  @field('producto_id') productoId!: string;
  @field('producto_server_id') productoServerId!: number | null;
  @field('cantidad') cantidad!: number;
  @field('precio_unitario') precioUnitario!: number;
  @field('subtotal') subtotal!: number;
  @field('impuesto') impuesto!: number;
  @field('total') total!: number;
  @field('version') version!: number;
  @readonly @date('created_at') createdAt!: Date;
  @date('updated_at') updatedAt!: Date;
}

File: libs/HandySuites.Domain/Entities/DetalleDevolucion.cs
using System.ComponentModel.DataAnnotations.Schema;
using HandySuites.Domain.Common;

namespace HandySuites.Domain.Entities;

/// <summary>
/// Linea individual de una DevolucionPedido. Linked al DetallePedido original
/// (cuando esta disponible) para tracking de devoluciones parciales por linea.
/// CASCADE on parent delete.
/// </summary>
[Table("DetalleDevoluciones")]
public class DetalleDevolucion : AuditableEntity
{
    [Column("id")]
    public int Id { get; set; }

    [Column("mobile_record_id")]
    public string? MobileRecordId { get; set; }

    [Column("devolucion_id")]
    public int DevolucionId { get; set; }

    /// <summary>Link a la linea original del pedido. Permite calcular cantidad_devuelta acumulada por linea.</summary>
    [Column("detalle_pedido_id")]
    public int? DetallePedidoId { get; set; }

    [Column("producto_id")]
    public int ProductoId { get; set; }

    [Column("cantidad")]
    public decimal Cantidad { get; set; }

    [Column("precio_unitario")]
    public decimal PrecioUnitario { get; set; }

    [Column("subtotal")]
    public decimal Subtotal { get; set; }

    [Column("impuesto")]
    public decimal Impuesto { get; set; }

    [Column("total")]
    public decimal Total { get; set; }

    // Navigation properties
    public DevolucionPedido Devolucion { get; set; } = null!;
    public DetallePedido? DetallePedido { get; set; }
    public Producto Producto { get; set; } = null!;
}
